waspc/src/Analyzer/Evaluator/EvaluationError.hs

```diff
@@ -0,0 +1,27 @@
+module Analyzer.Evaluator.EvaluationError
+  ( EvaluationError (..),
+  )
+where
+
+import Analyzer.Type (Type)
+
+data EvaluationError
+  = -- | "ExpectedType expected actual"
+    ExpectedType Type Type
```

Comment:
Huh, I think this is in most other languages called "UnexpectedType" :D? I get the reasoning behind it though, but it is a bit unusual when reading it for the first time. But ok, probably better to keep it consistent.